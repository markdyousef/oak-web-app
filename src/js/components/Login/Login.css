:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    width: 100%;
    height: 100%;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
}

.login {
    width: 400px;
    height: 300px;
    background-color: var(--main-white-color);
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    padding: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
}

.inputs {
    margin-top: 20px;
    width: 100%;
    display: flex;
    flex-direction: column;
    height: 120px;
    justify-content: space-between;
}

.buttons {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
}
