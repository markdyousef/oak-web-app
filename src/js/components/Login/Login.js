// @flow
import React, { Component, PropTypes } from 'react';
import Input from '../shared/Input';
import Button from '../shared/Button';

import css from './Login.css';

class Login extends Component {
    static propTypes = {
        loginUser: PropTypes.func.isRequired,
        router: PropTypes.shape({
            push: PropTypes.func
        }).isRequired
    };
    constructor() {
        super();
        this.onSubmit = this.onSubmit.bind(this);
        this.state = {
            email: '',
            password: '',
            message: null
        };
    }
    onSubmit() {
        const { loginUser, router } = this.props;
        const { email, password } = this.state;

        // clear messsage
        this.setState({ message: null });

        // email validation
        if (email.length < 6) {
            return this.setState({ message: 'please provide a valid email'});
        }
        // password validation
        if (password.length < 6) {
            return this.setState({ message: 'your password should be at least 6 characters' });
        }
        return loginUser(email, password)
            .then((res) => {
                // save token to localstorage
                localStorage.authToken = res.data.loginUser;
                router.push('/');
            })
            .catch(err => this.setState({ message: err.errors[0].message }))
    }
    render() {
        const { email, password, message } = this.state;
        const { router } = this.props;
        return (
            <div className={css.container}>
                <div className={css.login}>
                    <h1>Login</h1>
                    <div className={css.inputs}>
                        <Input
                            title="Email"
                            value={email}
                            onChange={value => this.setState({ email: value })}
                            type="email"
                            placeholder="Email"
                        />
                        <Input
                            title="Password"
                            value={password}
                            onChange={value => this.setState({ password: value })}
                            type="password"
                            placeholder="Password"
                        />
                    </div>
                    <div className={css.buttons}>
                        <Button
                            onClick={() => router.push('/signup')}
                            text="Create Account"
                            // type="transparent"
                        />
                        <Button
                            onClick={this.onSubmit}
                            text="Login"
                            type="primary"
                        />
                    </div>
                    <div className={css.message}>
                        {message &&
                            <div>
                                <h5>{message}</h5>
                            </div>
                        }
                    </div>
                </div>
            </div>
        );
    }
}

export default Login;
