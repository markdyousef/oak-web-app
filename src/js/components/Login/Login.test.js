import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

it('renders correctly', () => {
    const component = renderer.create(
        <Login />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
