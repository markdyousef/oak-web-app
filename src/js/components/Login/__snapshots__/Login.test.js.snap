exports[`test renders correctly 1`] = `
<section
  className="gmuKOB">
  <div
    className="khSIyb">
    <h1>
      Sign in to Cuest
    </h1>
    <div>
      <div>
        <h5
          className="dRMKoo">
          EMAIL
        </h5>
        <input
          className="bwXyts"
          onChange={[Function]}
          onKeyDown={null}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div>
        <h5
          className="dRMKoo">
          PASSWORD
        </h5>
        <input
          className="bwXyts"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
    </div>
    <div>
      <button
        className="exnCOy"
        onClick={[Function]}>
        Sign in
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16">
          <g>
            <path
              d="M10.9,2.8l4.8,4.8c0.1,0.1,0.1,0.2,0.1,0.4s-0.1,0.3-0.1,0.4l-4.8,4.8c-0.2,0.2-0.5,0.2-0.7,0s-0.2-0.5,0-0.7l4-4H0.6 C0.3,8.5,0.1,8.3,0.1,8s0.2-0.5,0.5-0.5h13.6l-4-4C10,3.3,10,3,10.2,2.8S10.7,2.6,10.9,2.8z" />
          </g>
        </svg>
      </button>
    </div>
    <div />
  </div>
  <div
    className="iJtnyx">
    <p>
      Forgot your password? 
      <a
        onClick={[Function]}
        style={Object {}}>
        Get a new one
      </a>
    </p>
  </div>
  <section
    className="fYstbg" />
</section>
`;
